test_server_info failing due to no good version provided

now that there is no redirector with hardcoded version and version is reported of dandi-archive itself we hit good old https://github.com/dandi/dandi-archive/issues/222 leading to

```
_______________________________ test_server_info _______________________________
dandi/tests/test_utils.py:321: in test_server_info
    assert Version(data["version"]) >= Version("1.2.0")
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/semantic_version/base.py:105: in __init__
    major, minor, patch, prerelease, build = self.parse(version_string, partial)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/semantic_version/base.py:311: in parse
    raise ValueError('Invalid version string: %r' % version_string)
E   ValueError: Invalid version string: '0+unknown'
```

https://github.com/dandi/dandi-archive/pull/1054 might be the one to resolve it for the main deployments but I don't know if it would apply to our dockerized server testing since it hardcodes to 0.0.0 in  https://github.com/dandi/dandi-archive/pull/1054/files#diff-8fbdfcdd92c90a7625fcf215e3b363869ff8ce725869824d28380b863498d35dR14

--- a/dandi/tests/test_utils.py
+++ b/dandi/tests/test_utils.py
@@ -311,6 +311,7 @@ else:
     using_docker = mark.skipif_no_network
 
 
+@pytest.mark.xfail(reason="https://github.com/dandi/dandi-archive/issues/1045")
 @pytest.mark.redirector
 @using_docker
 def test_server_info() -> None:

